src/com/goide/inspections/unresolved/GoIntroduceFunctionFix.java

```diff
@@ -0,0 +1,238 @@
+import java.util.Map;
+
+class GoIntroduceFunctionFix extends LocalQuickFixAndIntentionActionOnPsiElement implements HighPriorityAction {
+  private final String myName;
+  private static final String FAMILY_NAME = "Create function";
+  private static final String INTERFACE_TYPE = "interface{}";
+
+  public GoIntroduceFunctionFix(@NotNull PsiElement element, @NotNull String name) {
+    super(element);
+    myName = name;
+  }
+
+  @Override
+  public void invoke(@NotNull Project project,
+                     @NotNull PsiFile file,
+                     @Nullable("is null when called from inspection") Editor editor,
+                     @NotNull PsiElement startElement,
+                     @NotNull PsiElement endElement) {
+    if (editor == null) {
+      LOG.error("Cannot run quick fix without editor: " + getClass().getSimpleName(),
+                AttachmentFactory.createAttachment(file.getVirtualFile()));
+      return;
+    }
+
+    if (!(startElement instanceof GoCallExpr)) {
+      return;
+    }
+
+    GoCallExpr call = (GoCallExpr)startElement;
+    List<GoExpression> args = call.getArgumentList().getExpressionList();
+    GoType resultType = ContainerUtil.getFirstItem(GoTypeUtil.getExpectedTypes(call));
+    GoFunctionDeclaration function = createFunctionDeclaration(file, myName, args, resultType);
+
+    PsiElement anchor = PsiTreeUtil.getParentOfType(call, GoTopLevelDeclaration.class);
+    if (anchor == null) return;
+
+    function = (GoFunctionDeclaration)file.addAfter(function, anchor);
+    if (function == null) return;
+    function = CodeInsightUtilCore.forcePsiPostprocessAndRestoreElement(function);
+    if (function == null) return;
+
+    TemplateBuilderImpl builder = new TemplateBuilderImpl(function);
+    setupFunctionParameters(function, builder, args);
+    setupFunctionResult(function, builder);
+    GoBlock body = function.getBlock();
+    builder.setEndVariableAfter(body == null || body.getLbrace() == null ? function : body.getLbrace());
+
+    function = CodeInsightUtilCore.forcePsiPostprocessAndRestoreElement(function);
+    if (function == null) return;
+
+    Template template = builder.buildTemplate();
+    TextRange range = function.getTextRange();
+    editor.getCaretModel().moveToOffset(range.getStartOffset());
+    editor.getDocument().deleteString(range.getStartOffset(), range.getEndOffset());
+
+    startTemplate(editor, template, project);
+  }
+
+  @NotNull
+  private static List<String> checkTypes(@NotNull PsiFile file, List<GoType> types) {
+    List<String> resultTypes = ContainerUtil.newSmartList();
+    Map<String, GoImportSpec> importMap = ((GoFile)file).getImportedPackagesMap();
+    for (GoType type : types) {
+      resultTypes.add("i " + checkType(file, type, importMap));
+    }
+    return resultTypes;
+  }
+
+  @NotNull
+  private static String checkType(@NotNull PsiFile file, @Nullable GoType type, Map<String, GoImportSpec> importMap) {
+    if (type == null) return "interface{}";
```

Comment:
try to reuse `com.goide.GoDocumentationProvider#getTypePresentation` with an ability to substutute type-to-string converter via function argument